### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by jubaan**
Bubble sort Project 1 of the Odin Project Ready:

- bubble_sort - sort number
- bubble_sort_by - sort strings

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Hey @jubaan 
Nice try with the project. Sorry about my previous review. This PR is fine and I was able to do the review. Check my comments for changes to be made.

### Martin Afani

File: bubble_sort.rb
Comment (whiz25) [inline bubble_sort.rb]: - [x] bubble_sort method is implemented and working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
- [x] bubble_sort sorts an array of numbers in increasing order

**Comment by jubaan** [comment]
> ## Hey @jubaan
> 
> Nice try with the project. This PR is invalid because of wrong git-flow and so I can continue with the review. You can read more about git-flow [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/)
> ### [The Fix]
> 
>     * [ ]  create a development branch and make the PR from `bubble-sort` to the development branch.
> 
> 
> ### Martin Afani

Hi @whiz25, I appreciate your comments. Just a quick observation of the topic (Git-flow) you are referring to. This lecture comes after the submission of this project so at the time of doing this project wouldn't be on my knowledge.

If I follow the Ruby curriculum path this would be necessary after the submission and approval of this project.

**Comment by whiz25** [comment]
Yes @jubaan. I just realised that. Make the changes I have requested and send for another review. Applologies!

**Comment by jubaan** [comment]
@whiz25 Thank you a lot for your observations!

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress @jubaan 
I like your zeal to learn from this project. I tested your changes and they still don't pass the edge cases.
Check my suggestion in the comments.

### Martin Afani

File: bubble_sort.rb
Comment (whiz25) [reply bubble_sort.rb]: This does not pass yet;
Try this
``def bubble_sort(arr)
   n = arr.length
   loop do
    swapped = false
    (n - 1).times do |i|
      if arr[i] > arr[i + 1]
        arr[i], arr[i + 1] = arr[i + 1], arr[i] # swap the elements
        swapped = true
      end
    end
    break unless swapped
  end
  arr
end``

Comment (jubaan) [reply bubble_sort.rb]: Thank you for the suggestion @whiz25!

**Review by whiz25 — APPROVED** [review]
## Good job team :+1: 
You have made the requested changes and all edge cases pass. Merge and proceed.

### Martin Afani

**Pull request merged**
